### ISSUE TITLE
GPU Simulation seg fault

### ISSUE THREAD

**Issue by VivekGandhi54** [UWB-Biocomputing/Graphitti#239]
`AllVertices::summationMap_` is double deleted.

1. In a GPU simulation, the `summationMap_` is first allocated (AllVertices.cpp, ln 42).
2. This pointer is then set to a second `allVerticesDevice.summationMap_` (AllIFNeurons_d.cu, ln 63).
3. Consequently, the original `summationMap_` is never deleted, and causes a memory leak.
4. After the sim, `allVerticesDevice.summationMap_` is deleted (AllIFNeurons_d.cu, ln 112).
5. When `AllVertices` tries to delete `summationMap_`, it points to the same thing (which was already deleted). So, it causes a segmentation fault (AllVertices.cpp, ln 34).

**Comment by stiber** [comment]
I think `allVerticesDevice.summationMap_` should be a GPU-side pointer, while the other is a CPU-side pointer. Have these gotten mixed up?

**Comment by VivekGandhi54** [comment]
Yup, here is where the mixup happens. I'm not sure what the point of this line of code is, maybe it's supposed to be doing a deep copy?

AllIFNeurons_d.cu, ln 62-63:
```
// get device summation point address
summationMap_ = allVerticesDevice.summationMap_;
```

**Comment by stiber** [comment]
I think this would be a good issue to bundle with #136, as `AllIFNeuronsDeviceProperties` is a holdover from when we had separate neuron/synapse properties structs, instead of having them integrated into the main classes. And, in this case, we see that `AllIFNeurons` keeps its own copy of the GPU summation map array base address pointer.

I think we'll have someone come into the lab interested in GPU work, and bringing order to this disorder could be a good mechanism for climbing the learning curve, before moving us the next step along on the GPU side.

**Comment by jardiamj** [comment]
This was fixed in #344. Since the CPU side `summationMap_` was only used in GPU-related code, we decided not to allocate memory for it when compiled for CPU.

I think we can close this one.

**Comment by stiber** [comment]
I shall make it so.
